File: JUnitTestingLastAct/src/Singleton/SingletonTestCase.java
package Singleton;


public class SingletonTestCase {

    public static boolean booleanMessage(String message) {
        return Singleton.Instance.newMessage().equals(message);
    }

    public static boolean allCreatedInstanceIsSame(Singleton singleton1, Singleton singleton2) {
        return singleton1.equals(singleton2);
    }

    public static Singleton objectExists() {
        Singleton singleton = Singleton.Instance;
        return singleton;
    }

    public static boolean isSame() {
        Singleton singleton1 = Singleton.Instance;
        Singleton singleton2 = Singleton.Instance;
        return singleton1.equals(singleton2);
    }

}

File: KotlinTesting/src/Test/Creational/SingletonTestCase.kt
package Singleton

object SingletonTestCase {
    fun booleanMessage(message: String): Boolean {
        return Singleton.Instance!!.newMessage() == message
    }

    fun allCreatedInstanceIsSame(singleton1: Singleton, singleton2: Singleton): Boolean {
        return singleton1 == singleton2
    }

    fun objectExists(): Singleton? {
        return Singleton.Instance
    }

    val isSame: Boolean
        get() {
            val singleton1 = Singleton.Instance
            val singleton2 = Singleton.Instance
            return singleton1 == singleton2
        }
}